find_cmake_package_name.py

```diff
@@ -78,40 +74,91 @@ def find_path(conan_root, package_info):
     """
 
     package_name, version, repository, branch = package_info
-    package_path = "{}/{}/{}/{}/{}".format(conan_root, package_name, version,
-                                           repository, branch)
+    package_path = f"{conan_root}/{package_name}/{version}/{repository}/{branch}"
 
     if not path.isdir(package_path):
         print(
-            "Error: path name \"{}\" is not a directory.".format(package_path),
+            'Error: path name "{package_path}" is not a directory.',
             file=sys.stderr)
         exit(1)
     return package_path
 
 
-def find_configurations(package_path):
+config_suffix = re.compile('(C|-c)onfig.cmake$')
+
+
+def find_configuration(package_path):
     """Find CMake configuration files.
 
     package_path -- The directory path to start searching from.
     """
 
-    config_suffix = re.compile('(C|-c)onfig.cmake$')
-    f = list()
-    for _, _, files in walk(package_path):
-        f += list(filter(lambda file: config_suffix.search(file), files))
-    f = list(dict.fromkeys(f))
-    return list(map(lambda i: config_suffix.sub('', i), f))
+    configurations = list()
 
+    for root, _, files in walk(package_path):
+        configurations += list(
+            filter(lambda file: config_suffix.search(file), files))
```

Comment:
The signature for list addition doesn't require a `list` on the right, just something respecting the `iterable` protocol,  so here you can drop the `list()` caste which should save some performance.